src/ops.rs

```diff
@@ -510,6 +549,34 @@ pub fn shr_32bit(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) ->
     Ok(())
 }
 
+pub fn adc_8bit(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Result<(), VMError>{
+    let carry_add = if vm.flags.carry{
+        1
+    }else{
+        0
+    };
+    let prelim_sum = vm.get_arg(pipeline.args[0].location)?.u8_exact()?;
+    vm.set_arg(pipeline.args[0].location, SizedValue::Byte(prelim_sum + carry_add));    
```

Comment:
This should be explicit about the additions, ie, `prelim_sum.wrapping_add(carry_add)`. Otherwise if prelim_sum is 255 and carry is set it will give an overflow error in debug builds

Reply:
got it.